packages/cli-hooks/src/protocols.js

```diff
@@ -0,0 +1,119 @@
+import minimist from 'minimist';
+
+/**
+ * An interface encapsulating a specific set of communication rules that the
+ * SDK and the CLI both implement.
+ * @typedef {object} Protocol
+ * @property {string} name -
+ *   Label representing the name of the protocol as expected by the Slack CLI.
+ * @property {typeof console.log} log -
+ *   Logging utility to surface diagnostic info from the SDK or username code.
+ * @property {typeof console.error} error -
+ *   Logging utility to surface errors from the SDK or username code.
+ * @property {typeof console.warn} warn -
+ *   Logging utility to surface warnings from the SDK or username code.
+ * @property {(data: string) => void} respond -
+ *   Utility method for responding to hook invocations with stringified JSON.
+ * @property {() => string[]} [getCLIFlags] -
+ *   Retrieves all command-line flags related to the protocol implementation.
+ *   Most useful when child processes are being spawned by the SDK.
+ * @property {() => void} [install] -
+ *   Optional instructions for the protocol to install itself into the runtime.
+ *   Similar to stubbing functionality in mocking or testing setup utilties.
+ *   Ensures that the protocol expectations are met by userland and SDK code.
+ * @property {() => void} [uninstall] -
+ *   Optional instructions for the protocol to remove itself from the runtime.
+ */
+
+const DEFAULT_PROTOCOL = 'default';
+const MSG_BOUNDARY_PROTOCOL = 'message-boundaries';
+const SUPPORTED_NAMED_PROTOCOLS = [MSG_BOUNDARY_PROTOCOL];
+
+/**
+ * The baseline CLI-SDK protocol. All responses in this protocol go to stdout.
+ * The CLI combines both stdout and stderr to interpret the hook response.
+ * This simplistic protocol has inherent limitation: no logging diagnostics!
+ * @param {string[]} args - Command-line arguments passed to this process.
+ * @returns {Protocol} Specified communication rules for the SDK to follow.
+ */
+export function BaseProtocol(args) {
```

Comment:
NIT: should we name this `DefaultProtocol`?

Reply:
Certainly a more correct and clear name! Will make the update now :raised_hands: